transport/peeragent.go

```diff
@@ -0,0 +1,13 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peeragent.go -package=transporttest go.uber.org/yarpc/transport PeerAgent
+
+/*
+PeerAgent represents an interface for managing Peers across different outbounds.  A PeerList will
```

Comment:
Nit: Avoid wording comments in the form "X is an interface that does y". Instead you can just say "X does y". Being an interface is implied by its type declaration.